### PR TITLE
DHFPROD-8163: Cypress-redundantWaitUntils

### PR DESCRIPTION
**Pull request by csilva-ml**
#### Checklist: 
```diff
- Note: do not change the below
```

-  ##### Owner:

- [x] JIRA_ID included in all the commit messages
- [x] PR title is in the format JIRA_ID:Title
- [x] Rebase the branch with upstream
- [x] Squashed all commits into a single commit
- [x] Code passes ESLint tests
- [x] Added Tests
- [x] Run UI tests
  

- ##### Reviewer:

- [x] Reviewed Tests

### CODE REVIEW

File: marklogic-data-hub-central/ui/e2e/cypress/support/helper.tsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  const [hour, minute, seconds] = new Date().toLocaleTimeString("en-US").split(":");
+  const time = `${month}${date}${year}${hour}${minute}${seconds.replace(" ", "")}`;
+  return time;
+}
+/**
+     * Generate a UNIQUE Random name using date and time.
+     * @param {String} prefix Some prefix that references what it will be naming
+     * @returns String
+     */
+export function generateUniqueName(prefix: string) {
```
Comment (briantang) [inline marklogic-data-hub-central/ui/e2e/cypress/support/helper.tsx:16]: I like this a lot! Great Idea!

File: marklogic-data-hub-central/ui/e2e/cypress/integration/curation/curate/addCustomStepToFlow.spec.tsx [hunk 2]
```diff
@@ -16,49 +19,39 @@ describe("Add Custom step to a flow", () => {
...
     runPage.setFlowDescription(`test flow for adding custom step`);
     loadPage.confirmationOptions("Save").click();
-    cy.wait(2000);
-    cy.waitForAsyncRequest();
+
   });
 
   it("Add custom step from Run tile and Run the step", {defaultCommandTimeout: 120000}, () => {
-    cy.waitUntil(() => toolbar.getRunToolbarIcon().should("be.visible")).click({force: true});
+    cy.log("**Expand flow and add step**");
```
Comment (briantang) [inline marklogic-data-hub-central/ui/e2e/cypress/integration/curation/curate/addCustomStepToFlow.spec.tsx:45]: I noticed a couple of these logs throughout the changes. Is that intended to leave in?

Comment (csilva-ml) [reply marklogic-data-hub-central/ui/e2e/cypress/integration/curation/curate/addCustomStepToFlow.spec.tsx:45]: This is a great conversation I was hoping to have with you guys. They were intended, I think this practice would be really really helpful for us (this is a great practice I've seen in other projects). Of course not just randomly but through the whole test (This task for existing tests would take a lot of time, that's why I only limited myself to parts were the tests were breaking a lot). I would love to continue this conversation, but coming back to this case. If you think this isolated log does not make sense I can delete it. My intention it's to make debugging and code reading much more clear. Cypress shows this log in the left panel among the rest of the actions, it's a very powerful tool

Comment (briantang) [reply marklogic-data-hub-central/ui/e2e/cypress/integration/curation/curate/addCustomStepToFlow.spec.tsx:45]: OK sure! If it is intended and a good practice that helps, I think it doesn't hurt to start adding these logs.

**Comment by briantang** [comment]
@csilva-ml I believe there is one failure in this PR that appears related. Probably the confirmation click is looking for "OK" but it should be "Ok". Moving to Draft state for now.

**Review by briantang — APPROVED** [review]
Tests passed in previous run. Looks good.

**Review by Sanjeevani19 — APPROVED** [review]

**Pull request merged**